ci-operator/step-registry/ipi/conf/azure/ipi-conf-azure-commands.sh

```diff
@@ -20,36 +20,62 @@ function getVersion() {
   echo "${version}"
 }
 
+architecture=${OCP_ARCH:-"amd64"}
+if [[ "${CLUSTER_TYPE}" == "azure-arm64" ]]; then
+  architecture="arm64"
+fi
+
 CONFIG="${SHARED_DIR}/install-config.yaml"
 
 REGION="${LEASED_RESOURCE}"
 echo "Azure region: ${REGION}"
 
+if [[ "${COMPUTE_NODE_TYPE}" == "" ]]; then
+  if [ "${architecture}" = "arm64" ]; then
```

Comment:
Remove 33-39 as suggested by @vrutkovs and let's set the COMPUTE_NODE_TYPE var from the config

Reply:
@aleskandro  ipi-conf-azure is not just for arm64 test, setting it to "" in ref.yaml config file and giving default value for both amd64 and arm64 in shell file is more easy.   
how can we set two default value for one variable from config file?

Reply:
We can default to amd64 (mostly as there's already a lot of amd64 tests) and in arm64 test use a custom COMPUTE_NODE_TYPE.

Reply:
>then we will need to set custom COMPUTE_NODE_TYPE to every workflow for arm on azure

Yes

> if we keep the current code, we can set or not set.

Yes, that's the tradeoff here - you'll run less rehearsals as the step code doesn't change anymore and move configuration to the test and not the step-registry. 

>just like what we did for aws now

This is exactly why we should _stop_ doing that - more and more logic would be encoded in the step, causing more rehearsals which are irrelevant and making the logic more and more complicated.

Reply:
Ok, totally got your point, will fix it, thanks so much for your reply.